zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/IZkStateListener.java

```diff
@@ -58,4 +58,8 @@ public interface IZkStateListener {
    *             On any error.
    */
   void handleSessionEstablishmentError(final Throwable error) throws Exception;
+
+  default void handleStateChanged(KeeperState prevState, KeeperState curState) throws Exception {
```

Comment:
I think intention here is to have backward compatibility with old API until we move to new API which accepts both args. It would be good to have this added in comments.

Reply:
Not essentially. The intention here is how to use current ZkClient in ZKMetaClient implementation.